knime_extension/src/nodes/locationanalysis.py

```diff
@@ -633,3 +681,548 @@ def getIndex(s):
+@knext.node(
+    name="P-median Solver",
+    node_type=knext.NodeType.PREDICTOR,
+    icon_path=__NODE_ICON_PATH + "pmedian.png",
+    category=__category,
+    after="",
+)
+@knext.input_table(
+    name="Input demand data",
+    description="Table including a demand column and an optional distance column to the closest required facilities.",
+)
+@knext.input_table(
+    name="Input OD Matrix for candidate facilities",
+    description="Table consisting of columns (totaling n+1 columns) for demand ID and the distance matrix to the candidate facilities (n).",
+)
+@knext.output_table(
+    name="P-median result table",
+    description="Candidate columns name and chosen status(1/0)",
+)
+@knut.pulp_node_description(
+    short_description="Solve the P-median problem to minimize total weighted spatial costs.",
+    description="""The [P-median model](https://en.wikipedia.org/wiki/Facility_location_problem), one of the most widely used location models,
+    locates p facilities and allocates demand nodes to them in order to minimize the total weighted distance traveled. 
+
+    The node requires two inputs. The first (top) input should contain the  columns: demand ID and demand size. 
+    Optionally, it can also include a column for distance to closest required facilities. 
+    The second (bottom) input should only include the demand ID column which is identical to the one from the top input, 
+    along with columns representing the distance matrix to candidate facilities (n).It has a dimension of m x (n+1).
+
+    The P-Median problem will be solved using the PuLP package.
+    """,
+    references={
+        "Pulp.Solver": "https://coin-or.github.io/pulp/guides/how_to_configure_solvers.html",
+    },
+)
+class PmedianSolverNode:
+    id_left = knext.ColumnParameter(
+        "Demand ID column from demand table",
+        "The column indicating the demand location IDs.",
+        # port_index=0,
+        column_filter=knut.is_int_or_string,
+        include_row_key=False,
+        include_none_column=False,
+    )
+    populationid = knext.ColumnParameter(
+        "Demand size column",
+        "The column for demand size (e.g., population) ",
+        # port_index=0,
+        column_filter=knut.is_numeric,
+        include_row_key=False,
+        include_none_column=False,
+    )
+
+    requiredid = knext.ColumnParameter(
+        "Distance column for required facilities",
+        "The travel cost to the nearest required facilities ",
+        # port_index=0,
+        column_filter=knut.is_numeric,
+        include_row_key=False,
+        include_none_column=True,
+    )
+
+    id_right = knext.ColumnParameter(
+        "Demand ID column from distance matrix table",
+        "The column indicating the demand location IDs.",
+        port_index=1,
+        column_filter=knut.is_int_or_string,
+        include_row_key=False,
+        include_none_column=False,
+    )
+    pnumber = knext.IntParameter(
```

Comment:
p_number? and add validation of input parameter

Reply:
added